### ISSUE TITLE
Expand app window to fit 50 lines

### ISSUE THREAD

**Issue by forthrin** [joncampbell123/dosbox-x#633]
When I enable "50 lines mode" in a DOS program, the font shrinks to half its size while the app window stays the same. Is it possible to make the app window expand to fit 50 lines with full character height?

**Comment by emendelson** [comment]
When your DOS program switches to 50-line mode, it loads an 8x8 font from the emulated video BIOS. I don't know how to change that behavior, but (with the help of many people, including Jon Campbell) I've added to my project some text modes that use the standard VGA font but 80x43, 80x50, 80x60, 80x66, 132x25, 132x43, 132x50, and 132x60 modes. Here are the essentials: https://github.com/joncampbell123/dosbox-x/issues/576

**Comment by joncampbell123** [comment]
That is also how it's done on real hardware. The VGA BIOS or DOS utility loads a 8x8 font while keeping the 400-line alphanumeric text mode. Note that 8 lines x 50 text rows = 400 lines.

**Comment by forthrin** [comment]
@emendelson: How do I change `dosbox.conf`? to get 80x50 characters in a double height window without character squashing?

**Comment by emendelson** [comment]
**@forthrin**: You modify the code as I described in issue 576, and then you add this command to the [autoexec] section of dosbox.conf:

`50lines`

But of course this requires you to add the 50-line mode to int10_modes.cpp and the 50line command to shell.cpp.

**Comment by emendelson** [comment]
I think also it might be important to modify mouse.cpp to take account of the new video modes and specify new max x or y locations. I'll experiment with that later.

**Comment by emendelson** [comment]
Yes, you should probably add some lines to Mouse_NewVideoMode in mouse.cpp. For example, if you use my mode 76 for 50-line text, then add:

`case 0x76:`
		`mouse.max_y = 799;`
		`break;`

And similar settings for any other custom video mode.

**Comment by rderooy** [comment]
If all you want is text mode, just set `output=ttf` and if you want 80x50, just type this:

```dos
config -set output=ttf
mode con cols=80 lines=50
```

or start dosbox-x as such:
```bash
dosbox-x -set 'output=ttf' -c 'mode con cols=80 lines=50'
```

Or set the corresponding options in the dosbox-x config file

![image](https://github.com/joncampbell123/dosbox-x/assets/17701098/5c4999b6-d0a3-4453-b419-0a1ea2062cf5)

**Comment by forthrin** [comment]
Appears fixed
